### PR TITLE
drop deprecated json/yaml newSerializers, use json.NewSerializerWithO…

### PR DESCRIPTION
**Pull request by liyuerich**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

condidering json/yaml newSerializers funcs are deprecated with NewSerializerWithOptions(), the functions are adjusted to the same.

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/118125

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
```

### CODE REVIEW

**Comment by liyuerich** [comment]
@deads2k  @caesarxuchao 
any comment and question? please help to review and approve. thanks.

**Comment by liyuerich** [comment]
> @deads2k @caesarxuchao any comment and question? please help to review and approve. thanks.

David，Chao 
any comment?

**Comment by liyuerich** [comment]
@deads2k @caesarxuchao
any comment and question? please help to review and approve. thanks.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go [hunk 1]
```diff
@@ -858,14 +858,14 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 					MediaTypeType:    "application",
 					MediaTypeSubType: "json",
 					EncodesAsText:    true,
-					Serializer:       json.NewSerializer(json.DefaultMetaFactory, creator, typer, false),
-					PrettySerializer: json.NewSerializer(json.DefaultMetaFactory, creator, typer, true),
+					Serializer:       json.NewSerializerWithOptions(json.DefaultMetaFactory, creator, typer, json.SerializerOptions{Pretty: false}),
```
Comment (benluddy) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:861]: Nit (here and elsewhere):

```suggestion
					Serializer:       json.NewSerializerWithOptions(json.DefaultMetaFactory, creator, typer, json.SerializerOptions{}),
```

**Review by benluddy — COMMENTED** [review]
Looks OK other than a nit.

The new signature was introduced about five years ago. Is there any intention to ever remove the old functions? The overhead of maintaining them is practically zero, so I'm unsure if there is an upside to justify forcing everyone to update their programs.

If we want to avoid using the deprecated functions in Kubernetes without removing them altogether, I would expect to see some kind of presubmit linter to enforce that. Otherwise we will be here again in a couple years making a similar change.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go (hunk 1 above)
Comment (liyuerich) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:861]: thanks review , will fix it ASAP.

Comment (liyuerich) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go]: resolve， please review. thanks.

**Comment by liyuerich** [comment]
hi @benluddy @deads2k , please help to review due to 1.31 is released last week, I suppose it is ok to merge after 1.31.  please let me know if there is any comments. Thanks.

**Comment by liyuerich** [comment]
> hi @benluddy @deads2k , please help to review due to 1.31 is released last week, I suppose it is ok to merge after 1.31. please let me know if there is any comments. Thanks.

any comment？

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook_test.go [hunk 2]
```diff
@@ -38,14 +38,14 @@ import (
 	auditinternal "k8s.io/apiserver/pkg/apis/audit"
 	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
 	"k8s.io/apiserver/pkg/audit"
-	"k8s.io/client-go/tools/clientcmd/api/v1"
```
Comment (benluddy) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook_test.go:41]: Nit: Irrelevant change (from goimports?).

File: staging/src/k8s.io/client-go/rest/watch/decoder_test.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ import (
 	"testing"
 	"time"
 
-	"k8s.io/api/core/v1"
```
Comment (benluddy) [inline staging/src/k8s.io/client-go/rest/watch/decoder_test.go:26]: And here.

File: staging/src/k8s.io/client-go/rest/watch/encoder_test.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 	"io"
 	"testing"
 
-	"k8s.io/api/core/v1"
```
Comment (benluddy) [inline staging/src/k8s.io/client-go/rest/watch/encoder_test.go:24]: And here.

**Review by benluddy — COMMENTED** [review]
Looks good to me. Something added import names in several places that are not significant or related to this PR. Would you mind fixing those to keep the diff clean?

**Comment by liyuerich** [comment]
@deads2k David, Ben takes first pass, could you help to review and approve ? thanks.

**Pull request merged**